synapseclient/client.py

```diff
@@ -6373,20 +6373,17 @@ async def rest_get_async(
         Returns:
             JSON encoding of response
         """
-        try:
-            response = await self._rest_call_async(
-                "get",
-                uri,
-                None,
-                endpoint,
-                headers,
-                retry_policy,
-                requests_session_async_synapse,
-                **kwargs,
-            )
-            return self._return_rest_body(response)
-        except Exception:
-            self.logger.exception("Error in rest_get_async")
+        response = await self._rest_call_async(
```

Comment:
This error handling was preventing Synapse API errors from being surfaced to client users. All tests passed after making this change.